lib/usage/from.js

```diff
@@ -65,55 +65,110 @@ function actions(command, plain) {
 
 function options(command, plain) {
   const ui = cliui()
-  const style = !!plain ? noop : colorize
-  const {flags, interactive} = command.options
+  const required_with_ui = cliui()
+  const required_without_ui = cliui()
+  const requires_one_ui = cliui()
+  const style = plain ? noop : colorize
+  const {flags, interactive, requires_one} = command.options
 
   for (const [flag, config] of Object.entries(flags)) {
     if (flag === 'interactive' && !interactive) continue
     const type = typeOf(config.type)
-    ui.div({
-      text: util.format(
-        '%s--%s%s'
-      , config.shorthand
+    let flag_text = util.format(
+      '--%s%s <%s>'
+    , flag
+    , type === 'boolean'
+        ? `, --no-${flag}`
+        : ''
+    , style(type)
+    )
+    if (config.required) {
+      flag_text += ` ${chalk.bold('(required)')}`
+    }
+    ui.div(
+      {
+        text: config.shorthand
           ? `-${config.shorthand}, `
           : ''
-      , flag
-      , type === 'boolean'
-          ? `, --no-${flag}`
-          : ''
-      )
+      , padding: [0, 0, 0, 2]
+      , align: 'left'
+      , width: 6
+      }
+    , {
+        text: flag_text
+      , align: 'left'
+      , width: 40
+      }
+    , {
+        text: isUndef(config.default)
+          ? ''
+          : `[${chalk.bold(config.default)}]`
+      , align: 'left'
+      , width: 14
+      }
+    , {
+        text: config.description || ''
+      , align: 'left'
+      , width: Math.max(80, width((config.description || '').trim()))
+      }
+    )
+
+    if (config.required_with) {
+      required_with_ui.div({
+        text: style(`* ${flag}, ${config.required_with.join(', ')}`)
+      , padding: [0, 0, 0, 2]
+      })
+    }
+    if (config.required_without) {
+      required_without_ui.div({
+        text: style(`* ${flag}, ${config.required_without.join(', ')}`)
+      , padding: [0, 0, 0, 2]
+      })
+    }
+  }
+
+  if (requires_one) {
+    requires_one_ui.div({
+      text: style(`* ${requires_one.join(', ')}`)
     , padding: [0, 0, 0, 2]
-    , width: 40
-    }, {
-      text: `<${style(type)}>`
-    , align: 'left', width: 10
-    }, {
-      text: isUndef(config.default)
-        ? ''
-        : `[${chalk.bold(config.default)}]`
-    , align: 'left'
-    , width: 20
-    }, {
-      text: config.description || ''
-    , align: 'left'
-    , width: Math.max(80, width((config.description || '').trim()))
     })
   }
 
   ui.div({
-    text: [
-      `${style('<...>')}: input type`
-    , `${style('*')}: repeatable flags`
-    , `${style('[...]')}: default values`
-    ].join(' | ')
+    text: `${style('<...>')}: input type`
+  , align: 'left'
   , padding: [2, 0, 0, 2]
   })
+  ui.div({
+    text: `${style('*')}: repeatable flags`
+  , align: 'left'
+  , padding: [0, 0, 0, 2]
+  })
+  ui.div({
+    text: `${style('[...]')}: default values`
+  , align: 'left'
+  , padding: [0, 0, 1, 2]
+  })
+
```

Comment:
I kind of like this one all on 1 line. Mind if we keep that?

Reply:
You mean all on one line?  I always found that hard to read (and noisy), but I can undo it.

Reply:
Undone.  It does look better having less lines.  I think what bothers me is that it tells me about repeatable flags when I don't have any, and also that `<type>` doesn't really need explanation.  `[...]` also looks like a spread operator, but I get it :).  Maybe `[default]` would be better, but who knows.